scanner/cmd/scanner/main.go

```diff
@@ -195,18 +197,43 @@ func createBackends(ctx context.Context, cfg *config.Config) (*Backends, error)
 	return &b, nil
 }
 
-// Close closes all the backends used by the scanner.
-func (b *Backends) Close(ctx context.Context) {
+// getServices returns the list of the API services based on the backends provided.
+func getServices(b *Backends) []grpc.APIService {
+	var srvs []grpc.APIService
+	if b.Indexer != nil {
+		srvs = append(srvs, services.NewIndexerService(b.Indexer))
+	}
 	if b.Matcher != nil {
-		err := b.Matcher.Close(ctx)
-		if err != nil {
-			zlog.Error(ctx).Err(err).Msg("closing matcher")
+		// Set the index report getter to the remote indexer if available, otherwise the
+		// local indexer. A nil getter is ok, see implementation.
+		var getter indexer.ReportGetter
+		getter = b.Indexer
+		if b.RemoteIndexer != nil {
+			getter = b.RemoteIndexer
 		}
+		srvs = append(srvs, services.NewMatcherService(b.Matcher, getter))
 	}
-	if b.Indexer != nil {
-		err := b.Indexer.Close(ctx)
-		if err != nil {
-			zlog.Error(ctx).Err(err).Msg("closing indexer")
+	return srvs
+}
+
+// Close closes all the backends used by the scanner.
+func (b *Backends) Close(ctx context.Context) {
+	type Closeable interface {
```

Comment:
Typically in something like `io`, we see `ReadCloser`, `ReadWriteCloser`, etc. Want to change this to `Closer`, too? `Closeable` reminds me more of Java's passive voice while I think go tends to use a more active voice (this object is able to be closed vs this object closes).

Also this can be lowercase since it's not exposed anywhere else

Reply:
This is a local interface. Closeable is not only related to Java but also other languages. I understand where you're coming from, but this is fine.